makeStage2.sh

```diff
@@ -39,5 +40,66 @@ cd $usepop/pop/x/src/
 # ways to construct the Poplog system.
 # $usepop/pop/src/newpop -link -nox -norsv
 # $usepop/pop/src/newpop -link -x=-xm -norsv
+# $usepop/pop/src/newpop -link -x=-xt -norsv
+
+mkdir -p ${BUILD_HOME}/environments
+
+# We need to capture the Poplog environment for each build variant.
+echo_env() {
+    cmd='(usepop="'"$1"'" && . $usepop/pop/com/popenv.sh && env)'
+    env -i sh -c "$cmd" | sort | \
+    grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' | \
+    sed -e 's!'"$1"'![//USEPOP//]!g'
+}
+
+### Xt #########################################################################
+
 $usepop/pop/src/newpop -link -x=-xt -norsv
 
+echo_env "$usepop" > ${BUILD_HOME}/environments/xt
+echo_env "$usepop/pop/.." > ${BUILD_HOME}/environments/xt-cmp
+( cd ${BUILD_HOME}/environments && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-xt!g' < xt | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-xt/!g' > xt-new )
+
+mkdir -p $usepop/pop/pop-xt
+mkdir -p $usepop/pop/lib/psv-xt
+( cd $usepop/pop/pop; tar cf - . ) | ( cd $usepop/pop/pop-xt; tar xf - )
+( cd $usepop/pop/lib/psv; tar cf - . ) | ( cd $usepop/pop/lib/psv-xt; tar xf - )
+
+
+### motif ######################################################################
+
+$usepop/pop/src/newpop -link -x=-xm -norsv
```

Comment:
Can these `newpop` invocations be extracted to functions with descriptive names, e.g. `build_corepop_with_motif` or something along those lines, or maybe even just a single function that takes an arg specifying the type of build.

Reply:
`newpop` has its own extensive (and confusing) documentation in HELP NEWPOP. It's a swiss army knife that does many activities that can't easily be captured in a simple phrase other than 'rebuilds $popsys'. After a bit of thought I felt that a comment would be appropriate rather than trying to capture it in a single function name. Let me know if that goes far enough or not.

Reply:
That's fine, but I think it's worth sign posting to the docs then. A simple 'see HELP NEWPOP for details' would help someone unfamiliar like myself.

Reply:
Yeah I should have added that.

Reply:
Done